### PR TITLE
Create pprof download of profiles

### PR DESCRIPTION
**Pull request by metalmatze**
Reading profiles from storage and then converting them back to pprof format and open them as download! :tada:

This needs a rebase to work with the new UUID location IDs, but should be good to go otherwise.

### CODE REVIEW

File: proto/parca/query/v1alpha1/query.proto [hunk 1]
```diff
@@ -333,6 +340,11 @@ message QueryResponse{
     }
 }
 
+// QueryPprofResponse holds a pb.gz pprof profile for downloading
+message QueryPprofResponse {
```
Comment (brancz) [inline proto/parca/query/v1alpha1/query.proto:344]: The `Query` endpoint already has the ability to request the kind of response the user is requesting. I expected pprof to be just a different option there next to flamegraph/callgraph/top etc.

Comment (metalmatze) [reply proto/parca/query/v1alpha1/query.proto:344]: Oh, yes. That would make a lot of sense. :smile: :100:

File: pkg/storage/pprof.go [hunk 2]
```diff
@@ -89,6 +89,7 @@ func generatePprof(ctx context.Context, locationStore Locations, ip InstantProfi
 		id := n.LocationID()
 		_, seenLocation := locationByID[id]
 		if !seenLocation {
+			// TODO(metalmatze): Improve this by calling once with a slice of IDs
```
Comment (kakkoyun) [inline pkg/storage/pprof.go:92]: 👍

**Review by kakkoyun — APPROVED** [review]
LGTM 🚀 Awesome job!

**Pull request merged**